bpf/lib/maps.h

```diff
@@ -199,11 +199,11 @@ struct {
 
 #endif /* ENABLE_EGRESS_GATEWAY */
 
-#ifndef SKIP_CALLS_MAP
-static __always_inline void ep_tail_call(struct __ctx_buff *ctx,
-					 const __u32 index)
+static __always_inline void ep_tail_call(struct __ctx_buff *ctx __maybe_unused,
+					 const __u32 index __maybe_unused)
 {
+#ifndef SKIP_CALLS_MAP
```

Comment:
Short answer, with this hunk reverted:

```
$ make -C bpf
In file included from bpf_sock.c:16:
In file included from /home/joe/git/cilium/bpf/lib/identity.h:7:
/home/joe/git/cilium/bpf/lib/drop.h:83:2: error: implicit declaration of function 'ep_tail_call' [-Werror,-Wimplicit-function-declaration]
        ep_tail_call(ctx, CILIUM_CALL_DROP_NOTIFY);
        ^
1 error generated.
```

Longer answer is that `bpf/bpf_sock.c` now includes `bpf/lib/identity.h` for the remote-node check, and `bpf/lib/identity.h` includes `bpf/lib/{dbg,drop.h}`, which then tries to handle event notifications, and hence calls into this `bpf/lib/maps.h` to get the right tail call. We don't actually intend for the sock program to tail call anywhere here but the way the includes are set up, the compiler will complain about these implicit declarations early on, even if it's going to optimize all that code away later on.

Any thoughts on a better way to handle this case? We could conditionally wrap the includes in `bpf/lib/identity.h` with `if __ctx_is == __ctx_skb` to skip that import tree from the socket path :thinking:

Reply:
Hrm, actually, `__ctx_is` is still set to `__ctx_skb` from the socket paths so that last proposal doesn't work just yet, this doesn't seem right but I don't know if there would be other implications for changing `__ctx_is` for the socket progs case.

Reply:
I think I found an easier fix :sweat_smile: 
```
$ git diff
diff --git a/bpf/lib/identity.h b/bpf/lib/identity.h
index 96fd339e49..2b0a1a7d67 100644
--- a/bpf/lib/identity.h
+++ b/bpf/lib/identity.h
@@ -4,7 +4,7 @@
 #ifndef __LIB_IDENTITY_H_
 #define __LIB_IDENTITY_H_
 
-#include "drop.h"
 #include "dbg.h"
 
 /**
$ make -C bpf build_all -j6 &> /dev/null
$ echo $?
0
```

Reply:
Somehow I figured that `dbg.h` would also include `lib/maps.h` but apparently that's not the case! Thanks, will apply this.